discovery-provider/src/tasks/entity_manager/entities/developer_app.py

```diff
@@ -118,18 +119,17 @@ def validate_developer_app_tx(params: ManageEntityParameters, metadata):
         raise IndexingValidationError(
             f"Invalid {params.action} Developer App Transaction, user id is required and was not provided"
         )
-    if user_id not in params.existing_records[EntityType.USER]:
+    if user_id not in params.existing_records["User"]:
         raise IndexingValidationError(
             f"Invalid {params.action} Developer App Transaction, user id {user_id} does not exist"
         )
-    if not params.existing_records[EntityType.USER][user_id].wallet:
+    if not params.existing_records["User"][user_id].wallet:
```

Comment:
could be EntityType.USER.value if you wanna keep the enum since relying on the string literal could be a problem.

orrr define existing_records as Dict[EntityType, Dict] but leave the keys and values for that inner dict as any.

Reply:
re EntityType.USER.value:
this works, but it won't benefit from the type-hints the IDE will give, see
<img width="680" alt="Screenshot 2023-08-03 at 7 20 15 PM" src="https://github.com/AudiusProject/audius-protocol/assets/2731362/979eafb7-7d26-42d9-a015-60525c0db98e">


re Dict[EntityType, Dict] - this loses the typing of the inner things and forces you to have to try to typecast later on, which is definitely worse because of the any.


Rather than EntityType enum, a major shift in our client codebases we have discussed is using string literals instead of enums, and I think we'd benefit from that practice here some too

Reply:
I still lean team enum (I'll spare ya my elevator pitch for em) but seems there's not good IDE support here

Reply:
It's not so much IDE support, but more just support of the mypy type checker. I hear ya on the enums, but it also seems like there's no way to do type X = keyOf(y) like TS has.